services/web/server/src/simcore_service_webserver/socketio/handlers_utils.py

```diff
@@ -33,18 +37,34 @@ def has_socket_io_handler_signature(fun) -> bool:
 
 def register_handlers(app: web.Application, module: ModuleType):
     sio = get_socket_server(app)
-    predicate = (
-        lambda obj: inspect.isfunction(obj)
-        and has_socket_io_handler_signature(obj)
-        and inspect.iscoroutinefunction(obj)
-        and inspect.getmodule(obj) == module
-    )
-    member_fcts = inspect.getmembers(module, predicate)
+    member_fcts = [
+        fct for fct in socketio_handlers_registry if inspect.getmodule(fct) == module
+    ]
     # convert handler
     partial_fcts = [
-        socket_io_handler(app)(func_handler) for _, func_handler in member_fcts
+        socket_io_handler(app)(func_handler) for func_handler in member_fcts
     ]
     app[APP_CLIENT_SOCKET_DECORATED_HANDLERS_KEY] = partial_fcts
     # register the fcts
     for func in partial_fcts:
         sio.on(func.__name__, handler=func)
+
+
+def register_socketio_handler(func: callable) -> callable:
```

Comment:
MINOR: 
```python
from typing import Callable, Dict, Optional

HandlerT = Callable[[str, Dict, web.Application], Optional[bool]]

def register_socketio_handler(func: HandlerT) -> HandlerT:
```

And can even use ``HandlerT`` to assert signature below or moreover, perhaps ``mypy`` will catch issues when decorating functions of different signature